lib/reply.js

```diff
@@ -797,10 +797,11 @@ function notFound (reply) {
  * @param {object} context the request context
  * @param {object} data the JSON payload to serialize
  * @param {number} statusCode the http status code
+ * @param {string} contentType the reply content type
  * @returns {string} the serialized payload
  */
-function serialize (context, data, statusCode) {
-  const fnSerialize = getSchemaSerializer(context, statusCode)
+function serialize (context, data, statusCode, contentType) {
```

Comment:
This will require to also adjust the usage of the newest APIs that allows the usage of the serializers during the lifetime of a request. See: https://github.com/fastify/fastify/blob/main/lib/reply.js#L315-L398

Reply:
I will look into it later today